### DISCUSSION TITLE
Unable to update once apt-offline has sync'd packages

### DISCUSSION THREAD

**Discussion by virgilc73** [rickysarraf/apt-offline#197] [Q&A]
Greetings!
  First, thankyou for your work on the apt-offline project.  I am trying to use it to update Ubuntu 18.04 on some air-gapped machines and am having some issues getting it to work.  The following were my steps:
1.  Installed apt-offline on both on-line and off-line test machine.
    a.  apt-offline (version 1.8.1 all) installed to on-line machine using apt repository
    b.  apt offline (version 1.8.1 all) and dependencies installed using dpkg -i
2.  Created signature on offline machine for updates:  sudo apt-offline set --update apt-offline_update.sig
3.  Transferred the signature to online machine and grabbed repo update:  sudo apt-offline get --bundle update.zip apt-offline_update.sig
4. Copied update.zip to offline machine and ran:  sudo apt-offline install update.zip
    a.  Select Y to proceed with installation
    b.  Got ERROR: Path for keyring is invalid:  /etc/apt/trusted.gpg
         i.  I did not see this as an issue as all the the needed gpg files were in /etc/apt/trusted.gpg.d/
    c.  Looks like all the packages synced successfully
5.  Created signature on offline machine for upgrades: sudo apt-offline set --update --upgrade apt-offline_upgrade.sig
6.  Transferred the signature to online machine to get upgrades:  sudo apt-offline get --bundle upgrade.zip apt-offline_upgrade.sig
7.  Copied upgrade.zip to offline machine and ran:  sudo apt-offline install upgrade.zip
    a.  results were exactly the same as update process (I was expecting the bundles to be different sizes, but they were both 37.6MB)
8.  Once completed I ran apt upgrade and it looked normal, displaying the following:
    Reading package lists...Done
    Building dependency tree
    Reading stat information... Done
    Calculating upgrade... Done
    The following packages will be upgraded:
       Lotsa packages listed
    155 upgraded, 0 newly installed 0 to remove and 0 not upgraded
    137 standard security updates
    need to get 205 MB of archives.
    After this operation, 1,716 kB of additional disk space will be used.
    Do you want to continue? [Y/n] Y
9  I am now seeing the following errors for all of the packages:
    Ign:1 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 bash amd64 4.4.18-2ubuntu1.3
    Err:2 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libc6 amd64 2.27-3ubuntu1.6
      Temporary failure resolving 'archive.ubuntu.com'
    Err:1 http://security.ubuntu.com/ubuntu bionic-updates/main amd64 bash amd64 4.4.18-2ubuntu1.3
      Temporary failure resolving 'archive.ubuntu.com'
    Ign:3 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 cron amd64 3.0pl1-128.1ubuntu1.2
    Ign:4 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 tar amd64 1.29b-2ubuntu0.3
    Err:3 http://security.ubuntu.com/ubuntu bionic-updates/main amd64 cron amd64 3.0pl1-128.1ubuntu1.2
      Temporary failure resolving 'archive.ubuntu.com' 
      .....
10.  At the end of this process nothing is upgraded.

Based on this information is there anything you can suggest?  I have read the wiki and watched the youtube video, but this is still not working for me.

I also checked /var/cache/apt/archives and found that it does not contain any packages.  Is that were apt-offline is supposed to extract the packages?

**Comment by rickysarraf** [comment]
I know it is quite late in asking this but better ask than keep mum.

For Step 3, can you please share more details ? Like console output etc.
